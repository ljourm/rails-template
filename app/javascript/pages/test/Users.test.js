import { createLocalVue, mount } from "@vue/test-utils"
import VueRouter from "vue-router"
import Vuex from "vuex"
import Component from "../Users"

const email = "test@example.com"
const users = [{ uuid: "uuid", email: email, name: "name", roles: [] }]

describe("Component", () => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)
  localVue.use(Vuex)

  const router = new VueRouter()

  const state = { users: users }
  const store = new Vuex.Store({
    modules: {
      users: {
        namespaced: true,
        state,
      },
    },
  })

  const wrapper = mount(Component, {
    localVue,
    router,
    store,
  })

  it("snapshot", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
