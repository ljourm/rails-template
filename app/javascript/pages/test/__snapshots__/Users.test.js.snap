// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<div
  class="user-list"
  users="[object Object]"
>
  <h2>
    user-list
  </h2>
  <table
    class="table is-striped is-fullwidth"
  >
    <tr>
      <td>
        <a
          class=""
          href="#/users/uuid"
        >
          name
        </a>
      </td>
      <td>
        <a
          href="mailto:test@example.com"
        >
          test@example.com
        </a>
      </td>
      <td />
    </tr>
  </table>
</div>
`;
